T18: Механика игры на Websocket

artbuzzon opened the pull request:
Изменения в api оставил как техдолг (завел карточку в трелло). Там надо класс, работающий с апи править

artbuzzon committed:
workign version of server done

artbuzzon committed:
workign version of server done

artbuzzon committed:
first working version

artbuzzon committed:
first working version

artbuzzon committed:
mark up for game screen done

artbuzzon committed:
mark up for game screen done

artbuzzon committed:
auth refactored for id on the server

artbuzzon committed:
auth refactored for id on the server

artbuzzon committed:
chat done

artbuzzon committed:
chat done

artbuzzon commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Механика игры на Websocket](https://trello.com/c/7Q6Kg7Mi/18-%D0%BC%D0%B5%D1%85%D0%B0%D0%BD%D0%B8%D0%BA%D0%B0-%D0%B8%D0%B3%D1%80%D1%8B-%D0%BD%D0%B0-websocket)

alex1056 commented on server/index.ts:
```diff
@@ -0,0 +1,155 @@
+    next()
+}
+
+app.post('/create', (req, res) => {
```
Предлагаю сделать рутер и внутри него раскидывать запросы по урлам, т.е. в index будет только вызов: app.use('/', router); А внутри рутера вызывать свои контроллеры под маршрут. Сейчас в одном файле все. Немного не структурировано получается.

alex1056 commented on src/pages/game/game.tsx:
```diff
@@ -1,28 +1,203 @@
+//     const status = playerStatus === PLAYER_STATUSES.drawer ? PLAYER_STATUSES.guesser : PLAYER_STATUSES.drawer
+//     setPlayerStatus(status)
+// }
+
```
Наверное, лишнее

alex1056 commented on src/services/api-client.ts:
```diff
@@ -56,7 +56,6 @@ export class ApiClient {
     try {
       const res = await fetch(`${this._apiBase}${endpoint}/`, {
         method: 'POST',
-        credentials: 'include',
```
А без "credentials: 'include'"post запрос, например, для logout работать не будет. Если нужен специальный пост запрос - предлагаю его таким сделать (отдельно) назвать соответственно.

alex1056 commented on src/services/api.ts:
```diff
@@ -60,3 +61,53 @@ export const changePasswordRequest = async (
+/*
+    GAME API
+ */
+
```
Все что ниже предлагаю вынести в отдельный файл game-api.ts

alex1056 reviewed (commented):
структуризация и с post запросом надо уточнить, не ошибка ли.

alex1056 committed:
После pull, router->Router

alex1056 committed:
S6T25: Добавить Redux-Thunk (Saga) в запросы, S6T31: Добавляет API для изменения профиля пользователя. (#22)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* T11: Экран начала игры (#15)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* waiting for users component added

* cosmetics

* S5T22: Модальное окно (#12)

* modal started

* modal done

* cosmetics

* cosmetics

* S5T23: Add User component (#17)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S5T21: Компонент Header (навигация) (#14)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Header

* fix согласно комментариям

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T2: Главная страница, S5T13: Завершение игры (например, уведомление с кнопками «Повторить» или «Вернуться в главное меню»). (#13)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Добавлен компонент Layout

На главную страницу добавили форму авторизации

* Окончание игры: 2 кнопки - Повторить, Главное меню

* S5T22: Модальное окно (#12)

* modal started

* modal done

* cosmetics

* cosmetics

* S5T23: Add User component (#17)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S5T21: Компонент Header (навигация) (#14)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Header

* fix согласно комментариям

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* Fix after review

* sass loader added

* Донастройки конфигов + input версия 1

* Добавлен компонент Layout

На главную страницу добавили форму авторизации

* Окончание игры: 2 кнопки - Повторить, Главное меню

* Fix after review

Co-authored-by: Mike <mik.arbuzov@gmail.com>
Co-authored-by: artbuzzon <62429835+artbuzzon@users.noreply.github.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

* S5T3: Страница входа (#19)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Авторизация

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T8: forum & detail forum pages (#16)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S6T24: Добавить Redux-хранилище (#20)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Убрал лишнее

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T5: Страница пользователя (#18)

* Компонент Profile с детьми 100%

* Редактирование профиля убрал в Popup

* Fix согласно ревью

* S5T3: Страница входа (#19)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Авторизация

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T8: forum & detail forum pages (#16)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S6T24: Добавить Redux-хранилище (#20)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Убрал лишнее

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* Добавил useCallback

* Fixed conflicts

Co-authored-by: Mike <mik.arbuzov@gmail.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

* Fixed btn style

* waiting for users component added

* cosmetics

Co-authored-by: Alexander <gituser1056@gmail.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>
Co-authored-by: Alexander <64581128+alex1056@users.noreply.github.com>

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Fix после ревью

Co-authored-by: Mike <mik.arbuzov@gmail.com>
Co-authored-by: artbuzzon <62429835+artbuzzon@users.noreply.github.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

alex1056 committed:
Merge branch 'main' of github.com:SevillaTeam/crocodile

alex1056 committed:
S6T25: Fixed response with null data (#26)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* После pull, router->Router

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Fix после ревью

* Fixed with null data response

Co-authored-by: Mike <mik.arbuzov@gmail.com>

alex1056 committed:
Merge branch 'main' of github.com:SevillaTeam/crocodile

alex1056 committed:
Fixed err in store

artbuzzon committed:
api calls for game moved to a separate file, game server api paths moved to the router

artbuzzon committed:
api calls for game moved to a separate file, game server api paths moved to the router

alex1056 reviewed (approved):
Выглядит круто!

bercly0b commented on src/components/ChatInput/chat-input.tsx:
```diff
@@ -0,0 +1,39 @@
+
+export const ChatInput: FC<Props> = ({sendMessage}) => {
+
+  const submitMessage = (e: React.FormEvent<HTMLFormElement>) => {
```
`useCallback` хорошо бы заиспользовать

bercly0b commented on src/components/ChatInput/chat-input.tsx:
```diff
@@ -0,0 +1,39 @@
+
+  const [messageValue, setMessageValue] = React.useState('')
+
+  const changeMessageValue = ({ value }: IInputState) => {
```
и тут

bercly0b commented on src/pages/game/game.tsx:
```diff
@@ -1,28 +1,191 @@
+
+    const [chatMessages, setChatMessages] = React.useState([{content: '', username: ''}])
+
+    const connect = async () => {
```
тут можно было бы логику по работе с сетью вынести в отдельный файлик, что бы разгрузить компонент

bercly0b commented on src/services/game-api.ts:
```diff
@@ -0,0 +1,42 @@
+
+const baseUrl = 'http://localhost:8081'
```
такие вещи обычно в какой то конфиг выносят, что бы импортить из одного места. на случай если понадобится изменить порт или урл, или для другого окружения подложить другое значение.

bercly0b reviewed (approved):
в целом ок. пройдись пож по комментаим перед влитием

alex1056 committed:
Тесты на: input, button, modal (#28)

artbuzzon committed:
usecallback added

artbuzzon committed:
usecallback added

artbuzzon committed:
conflicts resolved
